{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2021 [Rosetta PyRosetta4.Release.python38.ubuntu 2021.18+release.54b4909cd528ede1d749ea69e8046c244fc797f2 2021-05-04T21:04:43] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "\u001b[0mcore.init: \u001b[0mChecking for fconfig files in pwd and ./rosetta/flags\n",
      "\u001b[0mcore.init: \u001b[0mRosetta version: PyRosetta4.Release.python38.ubuntu r282 2021.18+release.54b4909cd52 54b4909cd528ede1d749ea69e8046c244fc797f2 http://www.pyrosetta.org 2021-05-04T21:04:43\n",
      "\u001b[0mcore.init: \u001b[0mcommand: PyRosetta -ignore_unrecognized_res 1 -include_current -database /home/jwlee/.local/lib/python3.8/site-packages/pyrosetta/database\n",
      "\u001b[0mbasic.random.init_random_generator: \u001b[0m'RNG device' seed mode, using '/dev/urandom', seed=210568273 seed_offset=0 real_seed=210568273\n",
      "\u001b[0mbasic.random.init_random_generator: \u001b[0mRandomGenerator:init: Normal mode, seed=210568273 RG_type=mt19937\n"
     ]
    }
   ],
   "source": [
    "# Porting command line interface over to Jupyter notebook\n",
    "master_dir = 'stuff_for_jihoon_2021-06-04/'\n",
    "output_dir = master_dir + 'run4/'\n",
    "\n",
    "peptides_listfile = master_dir + 'peptides_list_Jihoon-run4-KRAS_10mer-mods.txt'\n",
    "pmhc_template_pdbfile = master_dir + '5yxn.pdb'\n",
    "weights_filetag = 'ref2015_cart.wts' # This is the default\n",
    "\n",
    "outfile_prefix = 'run4'\n",
    "\n",
    "### more imports\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import pyrosetta\n",
    "\n",
    "from pyrosetta.rosetta import core, protocols, numeric, basic, utility\n",
    "from pyrosetta.rosetta.utility import vector1_bool as bools\n",
    "\n",
    "# pyrosetta init\n",
    "init_flags = '-ignore_unrecognized_res 1 -include_current'\n",
    "\n",
    "#if args.ex1:\n",
    "#    init_flags += f' -ex1'\n",
    "#if args.ex2:\n",
    "#    init_flags += f' -ex2'\n",
    "\n",
    "pyrosetta.init(init_flags)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "################################################################################\n",
    "# FUNCTIONS\n",
    "################################################################################\n",
    "\n",
    "def mutate_peptide_sequence( peptide_sequence, pose ):\n",
    "    ''' This function mutates the current peptide sequence in pose\n",
    "    to peptide_sequence\n",
    "    '''\n",
    "    old_sequence = pose.chain_sequence(2)\n",
    "    print(f'mutate_peptide_sequence: from {old_sequence} to',\n",
    "          peptide_sequence)\n",
    "\n",
    "    pep_begin = pose.chain_begin(2)\n",
    "    pep_end = pose.chain_end(2)\n",
    "\n",
    "    amino_acids = 'ACDEFGHIKLMNPQRSTVWY'\n",
    "\n",
    "    if all(x in amino_acids for x in peptide_sequence): # all canonical AAs\n",
    "        assert len(peptide_sequence) == pep_end-pep_begin+1\n",
    "\n",
    "        tf = core.pack.task.TaskFactory()\n",
    "\n",
    "        # freeze non-peptide:\n",
    "        op = core.pack.task.operation.PreventRepacking()\n",
    "        for i in range(1,pose.size()+1):\n",
    "            if pose.chain(i) != 2:\n",
    "                op.include_residue(i)\n",
    "        tf.push_back(op)\n",
    "\n",
    "        # force desired sequence at peptide\n",
    "        for i, aa in zip(range(pose.chain_begin(2), pose.chain_end(2)+1),\n",
    "                         peptide_sequence):\n",
    "            op = core.pack.task.operation.RestrictAbsentCanonicalAAS()\n",
    "            op.include_residue(i)\n",
    "            op.keep_aas(aa)\n",
    "            tf.push_back(op)\n",
    "\n",
    "        packer = protocols.minimization_packing.PackRotamersMover()\n",
    "        packer.task_factory(tf)\n",
    "\n",
    "        packer.apply(pose)\n",
    "    else:\n",
    "        peptide_pose = core.pose.Pose()\n",
    "        core.pose.make_pose_from_sequence(\n",
    "            peptide_pose, peptide_sequence, \"fa_standard\")\n",
    "        for i in range(len(old_sequence)):\n",
    "            old_rsd = pose.residue(pep_begin+i)\n",
    "            new_rsd = peptide_pose.residue(1+i)\n",
    "            if old_rsd.name() != new_rsd.name():\n",
    "                print('residue change:', old_rsd.name(), new_rsd.name())\n",
    "                pose.replace_residue(pep_begin+i, new_rsd, True)\n",
    "\n",
    "\n",
    "def fastrelax_peptide(\n",
    "        scorefxn,\n",
    "        nrepeats,\n",
    "        peptide_positions,\n",
    "        neighbor_positions,\n",
    "        pose\n",
    "):\n",
    "    ''' This function optimizes the rosetta energy of the mhc-peptide complex\n",
    "    It allows the amino acid side chains to change and all atoms in the peptide\n",
    "    and nearby to move a small amount\n",
    "    '''\n",
    "    # movemap:\n",
    "    mm = pyrosetta.rosetta.core.kinematics.MoveMap()\n",
    "    mm.set_bb(False)\n",
    "    mm.set_chi(False)\n",
    "    mm.set_jump(False)\n",
    "\n",
    "    for i in peptide_positions:\n",
    "        mm.set_bb(i, True)\n",
    "        mm.set_chi(i, True)\n",
    "\n",
    "    for i in neighbor_positions:\n",
    "        mm.set_chi(i, True)\n",
    "\n",
    "    print('fastrelax_peptide:: adding chi flex at',\n",
    "          len([x for x in neighbor_positions\n",
    "               if x not in set(peptide_positions)]))\n",
    "\n",
    "    fr = protocols.relax.FastRelax(scorefxn_in=scorefxn,\n",
    "                                   standard_repeats=nrepeats)\n",
    "    fr.cartesian(True)\n",
    "    fr.set_movemap(mm)\n",
    "    fr.set_movemap_disables_packing_of_fixed_chi_positions(True)\n",
    "\n",
    "    # For non-Cartesian scorefunctions, use \"dfpmin_armijo_nonmonotone\"\n",
    "    fr.min_type(\"lbfgs_armijo_nonmonotone\")\n",
    "    fr.apply(pose)\n",
    "\n",
    "\n",
    "def unbind_peptide(pose):\n",
    "    ''' This function slides the peptide away from the MHC so we can\n",
    "    compute an unbound energy\n",
    "    '''\n",
    "    posl = range(1,pose.size()+1)\n",
    "\n",
    "    chains = np.array([pose.chain(x) for x in posl])\n",
    "    calphas = np.array([pose.residue(x).xyz(\"CA\") for x in posl])\n",
    "\n",
    "    mhc_cen = np.mean(calphas[chains==1], axis=0)\n",
    "    pep_cen = np.mean(calphas[chains==2], axis=0)\n",
    "\n",
    "    trans = pep_cen - mhc_cen\n",
    "    trans = numeric.xyzVector_double_t(trans[0], trans[1], trans[2])\n",
    "\n",
    "    j = pose.jump(1)\n",
    "    j.translation_along_axis(pose.conformation().upstream_jump_stub(1), trans,\n",
    "                             25.0)\n",
    "\n",
    "    pose.set_jump(1, j)\n",
    "\n",
    "\n",
    "\n",
    "def find_neighbors(core_positions, pose, heavyatom_distance_threshold = 6.0):\n",
    "    ''' This function finds MHC residues that are nearby the peptide\n",
    "    They will be allowed to move during the energy optimization\n",
    "    '''\n",
    "    # include all the 'core' positions as neighbors (of themselves, e.g.)\n",
    "    nbr_positions = set(core_positions)\n",
    "\n",
    "    posl = range(1, pose.size()+1)\n",
    "\n",
    "    for i in posl:\n",
    "        rsd1 = pose.residue(i)\n",
    "        if rsd1.is_virtual_residue():\n",
    "            continue\n",
    "        for j in core_positions:\n",
    "            rsd2 = pose.residue(j)\n",
    "            if ( rsd1.nbr_atom_xyz().distance_squared( rsd2.nbr_atom_xyz() ) <=\n",
    "                 ( rsd1.nbr_radius() + rsd2.nbr_radius() +\n",
    "                   heavyatom_distance_threshold )**2):\n",
    "                nbr_positions.add(i)\n",
    "                break\n",
    "    return nbr_positions\n",
    "\n",
    "\n",
    "def read_pmhc_pose(filename, special=None):\n",
    "    # 5yxn has the 5th chain as the \n",
    "    ''' This function reads a PDB file and deletes everything except the\n",
    "    MHC peptide binding domain and the peptide\n",
    "\n",
    "    Three possibilities for the input pdb:\n",
    "    1) it has 3 chains: MHC, B2M, peptide\n",
    "    2) it has 5 chains: MHC, B2M, peptide, TCRa, TCRb\n",
    "    3) it has 2 chains: MHC, peptide\n",
    "\n",
    "    '''\n",
    "    pmhc_pose = pyrosetta.pose_from_pdb(filename)\n",
    "\n",
    "    # delete second domain of MHC\n",
    "    if pmhc_pose.chain_end(1) >= 180:\n",
    "        pmhc_pose.delete_residue_range_slow(180, pmhc_pose.chain_end(1))\n",
    "        core.pose.add_variant_type_to_pose_residue(\n",
    "            pmhc_pose, core.chemical.UPPER_TERMINUS_VARIANT,\n",
    "            pmhc_pose.chain_end(1));\n",
    "\n",
    "    if special=='5yxn':\n",
    "        # TCRa, TCRb, MHC, B2M, peptide (doesn't follow the above order)\n",
    "        \n",
    "        # Delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(4), pmhc_pose.chain_end(4))\n",
    "        # Delete TCR\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(1), pmhc_pose.chain_end(2))\n",
    "    \n",
    "    elif pmhc_pose.num_chains()==3:\n",
    "        # MHC, B2M, peptide\n",
    "        # delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(2), pmhc_pose.chain_end(2))\n",
    "\n",
    "    elif pmhc_pose.num_chains() == 5:\n",
    "        # MHC. B2M, peptide, TCRa, TCRb\n",
    "        # delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(2), pmhc_pose.chain_end(2))\n",
    "\n",
    "        # delete TCR\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(3), pmhc_pose.size())\n",
    "\n",
    "    assert pmhc_pose.num_chains() == 2\n",
    "    return pmhc_pose"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Functions are defined, so now we use them below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.import_pose.import_pose: \u001b[0mFile 'stuff_for_jihoon_2021-06-04/5yxn.pdb' automatically determined to be of type PDB\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue SER:CtermProteinFull 192\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue ASP:CtermProteinFull 433\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CE  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NZ  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NE  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CZ  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NH1 on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NH2 on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CE  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NZ  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue VAL:CtermProteinFull 818\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 23 92\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 23 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 92 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 23 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 92 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 215 283\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 215 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 283 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 215 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 283 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 734 789\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 734 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 789 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 734 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 789 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 534 597\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 534 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 597 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 534 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 597 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 135 185\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 135 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 185 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 135 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 185 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 334 399\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 334 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 399 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 334 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 399 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 160 360\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 160 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 360 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 160 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 360 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 636 692\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 636 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 692 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 636 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 692 CYD\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 610 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 701 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 706 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 728 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 753 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 756 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 783 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 107 rotamers at 7 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: deleting jump 3\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: renumbering jump 4 to 3\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: deleting jump 1\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: renumbering jump 3 to 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mutate_peptide_sequence: from KLVALGINAV to KLVVVGAVGV\n",
      "\u001b[0mprotocols.minimization_packing.PackRotamersMover: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m undefined ScoreFunction -- creating a default one\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 13 rotamers at 10 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating PDInteractionGraph\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  234.695  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  234.695  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -216.613  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1819 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -380.013  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -372.113  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -476.627  0.26496  0.26496  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -476.627  0.26496  0.26496  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -358.998  0.26496  0.26496  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1646 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -362.578  0.26496  0.26496  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -354.833  0.26496  0.26496  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -399.378  0.2155  0.2155  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -399.378  0.2155  0.2155  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -314.846  0.2155  0.2155  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1593 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -314.846  0.2155  0.2155  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -308.18  0.2155  0.2155  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -316.205  0.189893  0.189893  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -316.205  0.189893  0.189893  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -207.662  0.189893  0.189893  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1530 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -207.686  0.189893  0.189893  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -215.616  0.164972  0.164972  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -215.616  -215.616  0.164972  0.164972\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -215.616  0.164972  0.164972  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -215.616  0.164972  0.164972  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -130.863  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -130.863  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -345.159  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1789 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -364.923  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -361.291  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -406.161  0.122022  0.122022  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -406.161  0.122022  0.122022  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -310.552  0.122022  0.122022  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1638 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -314.516  0.122022  0.122022  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -308.562  0.122022  0.122022  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -331.754  0.107776  0.107776  0.154\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -331.754  0.107776  0.107776  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -254.325  0.107776  0.107776  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1587 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -254.325  0.107776  0.107776  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -248.22  0.107776  0.107776  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -254.638  0.0984368  0.0984368  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -254.638  0.0984368  0.0984368  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -152.886  0.0984368  0.0984368  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1528 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -152.886  0.0984368  0.0984368  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -157.972  0.0893643  0.0893643  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -157.972  -157.972  0.0893643  0.0893643\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -157.972  0.0893643  0.0893643  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -157.972  0.0893643  0.0893643  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:NtermProteinFull]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -215.61553333270072\n",
      "unbound_score: -157.97163912046503\n",
      "unbound_score_frozen: -130.86321119900916\n",
      "binding_energy: KLVVVGAVGV -57.64389421223569\n",
      "binding_energy_frozen: KLVVVGAVGV -84.75232213369156\n",
      "bound_energies: KLVVVGAVGV  fa_atr: -1680.331 fa_rep:   233.894 fa_sol:  1060.807 fa_intra_rep:     3.217 fa_intra_sol_xover4:    59.747 lk_ball_wtd:   -22.921 fa_elec:  -495.096 hbond_sr_bb:   -73.897 hbond_lr_bb:  -100.256 hbond_bb_sc:   -52.070 hbond_sc:   -46.139 dslf_fa13:    -2.369 omega:    56.087 fa_dun:   599.674 p_aa_pp:   -32.827 yhh_planarity:     0.545 ref:    71.610 rama_prepro:    16.601 cart_bonded:   188.109\n",
      "unbound_energies: KLVVVGAVGV  fa_atr: -1606.425 fa_rep:   223.101 fa_sol:  1025.623 fa_intra_rep:     3.124 fa_intra_sol_xover4:    59.398 lk_ball_wtd:   -23.921 fa_elec:  -463.109 hbond_sr_bb:   -73.922 hbond_lr_bb:  -100.256 hbond_bb_sc:   -33.248 hbond_sc:   -49.685 dslf_fa13:    -2.371 omega:    55.992 fa_dun:   594.992 p_aa_pp:   -34.207 yhh_planarity:     0.397 ref:    71.610 rama_prepro:    16.226 cart_bonded:   178.710\n",
      "mutate_peptide_sequence: from KLVALGINAV to K[LYS:monomethylated]LVVVGAVGV\n",
      "residue change: LYS:NtermProteinFull LYS:monomethylated\n",
      "\u001b[0mcore.chemical.AtomICoor: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue LYS:monomethylated 276.  Returning BOGUS ID instead.\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing an atom: 276  H   that depends on a nonexistent polymer connection!\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m  --> generating it using idealized coordinates.\n",
      "residue change: ALA VAL\n",
      "residue change: LEU VAL\n",
      "residue change: ILE ALA\n",
      "residue change: ASN VAL\n",
      "residue change: ALA GLY\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CM to DB with theta0 = 2.01114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CM,NZ,1HZ to DB with theta0 = 1.85884 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CM,NZ,2HZ to DB with theta0 = 1.85924 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM,HCM1 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM,HCM2 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM,HCM3 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HCM1,CM,HCM2 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HCM1,CM,HCM3 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HCM2,CM,HCM3 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CM, to DB with d0 = 1.463 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM,HCM1, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM,HCM2, to DB with d0 = 1.09057 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM,HCM3, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  771.554  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  771.554  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -89.0131  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1837 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -376.613  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -369.11  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -471.954  0.323764  0.323764  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -471.954  0.323764  0.323764  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -347.042  0.323764  0.323764  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1645 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -352.486  0.323764  0.323764  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -344.136  0.323764  0.323764  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -393.567  0.268406  0.268406  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -393.567  0.268406  0.268406  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -306.61  0.268406  0.268406  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1588 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -306.654  0.268406  0.268406  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -299.797  0.268406  0.268406  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -311.186  0.222899  0.222899  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -311.186  0.222899  0.222899  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -201.607  0.222899  0.222899  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1527 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -201.633  0.222899  0.222899  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -210.481  0.195433  0.195433  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -210.481  -210.481  0.195433  0.195433\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -210.481  0.195433  0.195433  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -210.481  0.195433  0.195433  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -118.803  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -118.803  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -334.576  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1808 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -361.321  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -357.826  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -403.115  0.139007  0.139007  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -403.115  0.139007  0.139007  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -306.699  0.139007  0.139007  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1657 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -310.092  0.139007  0.139007  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -303.978  0.139007  0.139007  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -328.183  0.126571  0.126571  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -328.183  0.126571  0.126571  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -250.149  0.126571  0.126571  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1605 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -250.149  0.126571  0.126571  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -243.995  0.126571  0.126571  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -250.575  0.122846  0.122846  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -250.575  0.122846  0.122846  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -148.324  0.122846  0.122846  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1546 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -148.324  0.122846  0.122846  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -153.65  0.133268  0.133268  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -153.65  -153.65  0.133268  0.133268\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -153.65  0.133268  0.133268  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -153.65  0.133268  0.133268  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:monomethylated]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -210.48140147747705\n",
      "unbound_score: -153.64995345232\n",
      "unbound_score_frozen: -118.80345452049636\n",
      "binding_energy: K[LYS:monomethylated]LVVVGAVGV -56.831448025157044\n",
      "binding_energy_frozen: K[LYS:monomethylated]LVVVGAVGV -91.67794695698069\n",
      "bound_energies: K[LYS:monomethylated]LVVVGAVGV  fa_atr: -1683.920 fa_rep:   235.380 fa_sol:  1055.194 fa_intra_rep:     3.237 fa_intra_sol_xover4:    59.984 lk_ball_wtd:   -20.268 fa_elec:  -500.773 hbond_sr_bb:   -73.891 hbond_lr_bb:  -100.256 hbond_bb_sc:   -48.165 hbond_sc:   -48.395 dslf_fa13:    -2.368 omega:    56.851 fa_dun:   604.733 p_aa_pp:   -31.858 yhh_planarity:     0.472 ref:    71.610 rama_prepro:    22.991 cart_bonded:   188.961\n",
      "unbound_energies: K[LYS:monomethylated]LVVVGAVGV  fa_atr: -1610.505 fa_rep:   224.178 fa_sol:  1030.472 fa_intra_rep:     3.152 fa_intra_sol_xover4:    59.747 lk_ball_wtd:   -22.224 fa_elec:  -473.270 hbond_sr_bb:   -73.902 hbond_lr_bb:  -100.256 hbond_bb_sc:   -33.025 hbond_sc:   -50.526 dslf_fa13:    -2.369 omega:    56.128 fa_dun:   598.238 p_aa_pp:   -33.259 yhh_planarity:     0.339 ref:    71.610 rama_prepro:    22.555 cart_bonded:   179.266\n",
      "mutate_peptide_sequence: from KLVALGINAV to K[LYS:dimethylated]LVVVGAVGV\n",
      "residue change: LYS:NtermProteinFull LYS:dimethylated\n",
      "\u001b[0mcore.chemical.AtomICoor: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue LYS:dimethylated 276.  Returning BOGUS ID instead.\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing an atom: 276  H   that depends on a nonexistent polymer connection!\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m  --> generating it using idealized coordinates.\n",
      "residue change: ALA VAL\n",
      "residue change: LEU VAL\n",
      "residue change: ILE ALA\n",
      "residue change: ASN VAL\n",
      "residue change: ALA GLY\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CH1 to DB with theta0 = 1.9621 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CH2 to DB with theta0 = 1.9621 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH1,NZ,CH2 to DB with theta0 = 1.85648 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH1,NZ,1HZ to DB with theta0 = 1.88467 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH2,NZ,1HZ to DB with theta0 = 1.88467 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH1,HH11 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH1,HH12 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH1,HH13 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH11,CH1,HH12 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH11,CH1,HH13 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH12,CH1,HH13 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH2,HH21 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH2,HH22 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH2,HH23 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH21,CH2,HH22 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH21,CH2,HH23 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH22,CH2,HH23 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CH1, to DB with d0 = 1.474 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CH2, to DB with d0 = 1.474 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH1,HH11, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH1,HH12, to DB with d0 = 1.09058 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH1,HH13, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH2,HH21, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH2,HH22, to DB with d0 = 1.09058 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH2,HH23, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  774.832  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  774.832  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -88.7312  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1865 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -385.011  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -377.534  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -483.523  0.313318  0.313318  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -483.523  0.313318  0.313318  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -357.754  0.313318  0.313318  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1664 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -361.617  0.313318  0.313318  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -353.076  0.313318  0.313318  0.154\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -400.637  0.26369  0.26369  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -400.637  0.26369  0.26369  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -311.562  0.26369  0.26369  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1611 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -311.562  0.26369  0.26369  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -304.538  0.26369  0.26369  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -315.164  0.22685  0.22685  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -315.164  0.22685  0.22685  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -203.61  0.22685  0.22685  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1554 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -203.635  0.22685  0.22685  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -214.268  0.165217  0.165217  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -214.268  -214.268  0.165217  0.165217\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -214.268  0.165217  0.165217  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -214.268  0.165217  0.165217  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -123.005  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -123.005  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -338.584  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1829 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -362.9  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -359.105  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -404.581  0.117837  0.117837  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -404.581  0.117837  0.117837  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -304.004  0.117837  0.117837  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1677 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -308.919  0.117837  0.117837  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -302.72  0.117837  0.117837  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -328.615  0.102578  0.102578  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -328.615  0.102578  0.102578  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -250.062  0.102578  0.102578  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1631 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -250.062  0.102578  0.102578  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -243.868  0.102578  0.102578  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -250.682  0.0998464  0.0998464  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -250.682  0.0998464  0.0998464  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -148.153  0.0998464  0.0998464  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1572 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -148.153  0.0998464  0.0998464  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -153.334  0.100289  0.100289  0.55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -153.334  -153.334  0.100289  0.100289\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -153.334  0.100289  0.100289  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -153.334  0.100289  0.100289  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:dimethylated]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -214.26774657581936\n",
      "unbound_score: -153.33414681986656\n",
      "unbound_score_frozen: -123.00490979638855\n",
      "binding_energy: K[LYS:dimethylated]LVVVGAVGV -60.9335997559528\n",
      "binding_energy_frozen: K[LYS:dimethylated]LVVVGAVGV -91.26283677943081\n",
      "bound_energies: K[LYS:dimethylated]LVVVGAVGV  fa_atr: -1684.561 fa_rep:   236.047 fa_sol:  1053.040 fa_intra_rep:     3.520 fa_intra_sol_xover4:    59.585 lk_ball_wtd:   -20.268 fa_elec:  -501.184 hbond_sr_bb:   -73.894 hbond_lr_bb:  -100.256 hbond_bb_sc:   -47.854 hbond_sc:   -47.577 dslf_fa13:    -2.369 omega:    56.668 fa_dun:   602.807 p_aa_pp:   -31.862 yhh_planarity:     0.575 ref:    71.610 rama_prepro:    22.894 cart_bonded:   188.813\n",
      "unbound_energies: K[LYS:dimethylated]LVVVGAVGV  fa_atr: -1610.268 fa_rep:   224.466 fa_sol:  1028.537 fa_intra_rep:     3.295 fa_intra_sol_xover4:    59.424 lk_ball_wtd:   -22.233 fa_elec:  -471.635 hbond_sr_bb:   -73.906 hbond_lr_bb:  -100.256 hbond_bb_sc:   -32.341 hbond_sc:   -50.592 dslf_fa13:    -2.373 omega:    56.005 fa_dun:   598.622 p_aa_pp:   -33.218 yhh_planarity:     0.369 ref:    71.610 rama_prepro:    22.413 cart_bonded:   178.747\n",
      "mutate_peptide_sequence: from KLVALGINAV to K[LYS:lys_trimethylated]LVVVGAVGV\n",
      "residue change: LYS:NtermProteinFull LYS:lys_trimethylated\n",
      "\u001b[0mcore.chemical.AtomICoor: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue LYS:lys_trimethylated 276.  Returning BOGUS ID instead.\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing an atom: 276  H   that depends on a nonexistent polymer connection!\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m  --> generating it using idealized coordinates.\n",
      "residue change: ALA VAL\n",
      "residue change: LEU VAL\n",
      "residue change: ILE ALA\n",
      "residue change: ASN VAL\n",
      "residue change: ALA GLY\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CM1 to DB with theta0 = 1.92318 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CM2 to DB with theta0 = 1.92318 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CM3 to DB with theta0 = 1.92318 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CM1,NZ,CM2 to DB with theta0 = 1.89792 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CM1,NZ,CM3 to DB with theta0 = 1.89792 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CM2,NZ,CM3 to DB with theta0 = 1.89792 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM1,HM11 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM1,HM12 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM1,HM13 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM11,CM1,HM12 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM11,CM1,HM13 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM12,CM1,HM13 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM2,HM21 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM2,HM22 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM2,HM23 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM21,CM2,HM22 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM21,CM2,HM23 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM22,CM2,HM23 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM3,HM31 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM3,HM32 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CM3,HM33 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM31,CM3,HM32 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM31,CM3,HM33 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HM32,CM3,HM33 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CM1, to DB with d0 = 1.482 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CM2, to DB with d0 = 1.482 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CM3, to DB with d0 = 1.482 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM1,HM11, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM1,HM12, to DB with d0 = 1.09058 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM1,HM13, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM2,HM21, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM2,HM22, to DB with d0 = 1.09058 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM2,HM23, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM3,HM31, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM3,HM32, to DB with d0 = 1.09058 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CM3,HM33, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  806.645  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  806.645  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -87.4341  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1835 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -387.065  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -378.966  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -487.141  0.338842  0.338842  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -487.141  0.338842  0.338842  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -353.658  0.338842  0.338842  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1638 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -358.121  0.338842  0.338842  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -349.256  0.338842  0.338842  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -400.965  0.26825  0.26825  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -400.965  0.26825  0.26825  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -310.627  0.26825  0.26825  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1582 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -310.627  0.26825  0.26825  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -303.503  0.26825  0.26825  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -314.73  0.230947  0.230947  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -314.73  0.230947  0.230947  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -202.592  0.230947  0.230947  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1522 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -202.841  0.230947  0.230947  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -216.107  0.175441  0.175441  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -216.107  -216.107  0.175441  0.175441\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -216.107  0.175441  0.175441  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -216.107  0.175441  0.175441  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -126.601  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -126.601  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -342.453  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1805 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -364.735  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -359.892  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -408.133  0.120984  0.120984  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -408.133  0.120984  0.120984  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -307.309  0.120984  0.120984  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1651 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -310.709  0.120984  0.120984  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -304.561  0.120984  0.120984  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -330.601  0.128584  0.128584  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -330.601  0.128584  0.128584  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -251.626  0.128584  0.128584  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1598 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -251.626  0.128584  0.128584  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -245.399  0.128584  0.128584  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -252.329  0.124897  0.124897  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -252.329  0.124897  0.124897  0.31955\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -149.501  0.124897  0.124897  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1539 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -149.501  0.124897  0.124897  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -155.785  0.112639  0.112639  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -155.785  -155.785  0.112639  0.112639\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -155.785  0.112639  0.112639  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -155.785  0.112639  0.112639  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:lys_trimethylated]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -216.10729171093203\n",
      "unbound_score: -155.78478291486442\n",
      "unbound_score_frozen: -126.60107976141569\n",
      "binding_energy: K[LYS:lys_trimethylated]LVVVGAVGV -60.32250879606761\n",
      "binding_energy_frozen: K[LYS:lys_trimethylated]LVVVGAVGV -89.50621194951634\n",
      "bound_energies: K[LYS:lys_trimethylated]LVVVGAVGV  fa_atr: -1683.835 fa_rep:   235.568 fa_sol:  1045.042 fa_intra_rep:     4.049 fa_intra_sol_xover4:    59.538 lk_ball_wtd:   -20.238 fa_elec:  -497.540 hbond_sr_bb:   -73.886 hbond_lr_bb:  -100.256 hbond_bb_sc:   -47.105 hbond_sc:   -48.705 dslf_fa13:    -2.370 omega:    56.829 fa_dun:   604.931 p_aa_pp:   -32.123 yhh_planarity:     0.472 ref:    71.610 rama_prepro:    22.834 cart_bonded:   189.077\n",
      "unbound_energies: K[LYS:lys_trimethylated]LVVVGAVGV  fa_atr: -1613.343 fa_rep:   224.831 fa_sol:  1027.496 fa_intra_rep:     3.404 fa_intra_sol_xover4:    59.296 lk_ball_wtd:   -22.715 fa_elec:  -471.198 hbond_sr_bb:   -73.941 hbond_lr_bb:  -100.256 hbond_bb_sc:   -32.336 hbond_sc:   -50.551 dslf_fa13:    -2.368 omega:    56.005 fa_dun:   599.842 p_aa_pp:   -33.550 yhh_planarity:     0.339 ref:    71.610 rama_prepro:    22.183 cart_bonded:   179.468\n",
      "mutate_peptide_sequence: from KLVALGINAV to K[LYS:acetylated]LVVVGAVGV\n",
      "residue change: LYS:NtermProteinFull LYS:acetylated\n",
      "\u001b[0mcore.chemical.AtomICoor: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m IcoorAtomID::atom_id(): Cannot get atom_id for POLYMER_LOWER of residue LYS:acetylated 276.  Returning BOGUS ID instead.\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing an atom: 276  H   that depends on a nonexistent polymer connection!\n",
      "\u001b[0mcore.conformation.Residue: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m  --> generating it using idealized coordinates.\n",
      "residue change: ALA VAL\n",
      "residue change: LEU VAL\n",
      "residue change: ILE ALA\n",
      "residue change: ASN VAL\n",
      "residue change: ALA GLY\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CE,NZ,CH to DB with theta0 = 2.10644 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH,NZ,1HZ to DB with theta0 = 1.80795 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH,CH3 to DB with theta0 = 2.02895 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: NZ,CH,OH to DB with theta0 = 2.08514 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH3,CH,OH to DB with theta0 = 2.1691 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH,CH3,HH31 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH,CH3,HH32 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: CH,CH3,HH33 to DB with theta0 = 1.91114 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH31,CH3,HH32 to DB with theta0 = 1.90998 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH31,CH3,HH33 to DB with theta0 = 1.90964 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined angle LYS: HH32,CH3,HH33 to DB with theta0 = 1.91077 , Ktheta = 80\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: NZ,CH, to DB with d0 = 1.375 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH,CH3, to DB with d0 = 1.509 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH,OH, to DB with d0 = 1.277 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH3,HH31, to DB with d0 = 1.09029 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH3,HH32, to DB with d0 = 1.09057 , Kd = 300\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mAdding undefined length LYS: CH3,HH33, to DB with d0 = 1.08906 , Kd = 300\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  1562.64  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  1562.64  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -52.4806  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1933 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -379.891  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -372.006  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -481.357  0.34816  0.34816  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -481.357  0.34816  0.34816  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -351.164  0.34816  0.34816  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1719 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -353.606  0.34816  0.34816  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -345.13  0.34816  0.34816  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -400.703  0.257744  0.257744  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -400.703  0.257744  0.257744  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -313.845  0.257744  0.257744  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1667 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -313.845  0.257744  0.257744  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -306.996  0.257744  0.257744  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -316.295  0.212981  0.212981  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -316.295  0.212981  0.212981  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -206.312  0.212981  0.212981  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1593 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -206.362  0.212981  0.212981  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -214.463  0.181245  0.181245  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -214.463  -214.463  0.181245  0.181245\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -214.463  0.181245  0.181245  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -214.463  0.181245  0.181245  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -126.276  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -126.276  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -341.367  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1898 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -361.231  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -357.725  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -403.091  0.121988  0.121988  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -403.091  0.121988  0.121988  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -307.576  0.121988  0.121988  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1754 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -310.954  0.121988  0.121988  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -304.913  0.121988  0.121988  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -328.035  0.108295  0.108295  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -328.035  0.108295  0.108295  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -250.354  0.108295  0.108295  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1693 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -250.354  0.108295  0.108295  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -244.228  0.108295  0.108295  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -250.839  0.0982063  0.0982063  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -250.839  0.0982063  0.0982063  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -149.066  0.0982063  0.0982063  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1634 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -149.066  0.0982063  0.0982063  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -154.166  0.091492  0.091492  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -154.166  -154.166  0.091492  0.091492\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -154.166  0.091492  0.091492  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -154.166  0.091492  0.091492  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:acetylated]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -214.46329957213743\n",
      "unbound_score: -154.16630494314444\n",
      "unbound_score_frozen: -126.27563116548765\n",
      "binding_energy: K[LYS:acetylated]LVVVGAVGV -60.29699462899299\n",
      "binding_energy_frozen: K[LYS:acetylated]LVVVGAVGV -88.18766840664978\n",
      "bound_energies: K[LYS:acetylated]LVVVGAVGV  fa_atr: -1685.775 fa_rep:   236.316 fa_sol:  1057.988 fa_intra_rep:     3.250 fa_intra_sol_xover4:    59.786 lk_ball_wtd:   -23.454 fa_elec:  -496.166 hbond_sr_bb:   -73.892 hbond_lr_bb:  -100.256 hbond_bb_sc:   -49.202 hbond_sc:   -47.975 dslf_fa13:    -2.380 omega:    56.170 fa_dun:   599.007 p_aa_pp:   -31.982 yhh_planarity:     0.714 ref:    71.610 rama_prepro:    22.567 cart_bonded:   189.209\n",
      "unbound_energies: K[LYS:acetylated]LVVVGAVGV  fa_atr: -1605.273 fa_rep:   223.149 fa_sol:  1021.808 fa_intra_rep:     3.121 fa_intra_sol_xover4:    59.573 lk_ball_wtd:   -23.011 fa_elec:  -465.454 hbond_sr_bb:   -73.908 hbond_lr_bb:  -100.256 hbond_bb_sc:   -33.252 hbond_sc:   -50.974 dslf_fa13:    -2.372 omega:    56.062 fa_dun:   595.898 p_aa_pp:   -33.141 yhh_planarity:     0.437 ref:    71.610 rama_prepro:    22.400 cart_bonded:   179.415\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DONE\n"
     ]
    }
   ],
   "source": [
    "pmhc_pose = read_pmhc_pose(pmhc_template_pdbfile, special='5yxn')\n",
    "\n",
    "# find neighbor positions\n",
    "peptide_positions = list(range(pmhc_pose.chain_begin(2),\n",
    "                               pmhc_pose.chain_end(2)+1))\n",
    "neighbor_positions = find_neighbors(peptide_positions, pmhc_pose)\n",
    "\n",
    "\n",
    "template_peptide = pmhc_pose.chain_sequence(2)\n",
    "\n",
    "with open(peptides_listfile, 'r') as data:\n",
    "    peptides = [x[:-1] for x in data]\n",
    "\n",
    "scorefxn = pyrosetta.create_score_function(weights_filetag)\n",
    "peptide_scores = []\n",
    "\n",
    "for peptide in peptides:\n",
    "    #if len(peptide) != len(template_peptide)\n",
    "    #: This is already checked in the mutate_peptide_sequence function\n",
    "    #    print(f'ERROR: cant model peptide of mismatched length:',\n",
    "    #          f'{peptide} on {template_peptide}')\n",
    "    #    continue\n",
    "\n",
    "    pose = pmhc_pose.clone()\n",
    "\n",
    "    mutate_peptide_sequence(peptide, pose)\n",
    "\n",
    "    fastrelax_peptide(scorefxn, 1, peptide_positions, neighbor_positions, pose)\n",
    "    bound_score = scorefxn(pose)\n",
    "\n",
    "    unbound_pose = pose.clone()\n",
    "\n",
    "    unbind_peptide(unbound_pose)\n",
    "    unbound_score_frozen = scorefxn(unbound_pose)\n",
    "\n",
    "    fastrelax_peptide(scorefxn, 1, peptide_positions, neighbor_positions,\n",
    "                      unbound_pose)\n",
    "    unbound_score = scorefxn(unbound_pose)\n",
    "\n",
    "    binding_energy = bound_score - unbound_score\n",
    "    binding_energy_frozen = bound_score - unbound_score_frozen\n",
    "\n",
    "\n",
    "    print('annotated_sequence:', pose.annotated_sequence())\n",
    "    print('bound_score:', bound_score)\n",
    "    print('unbound_score:', unbound_score)\n",
    "    print('unbound_score_frozen:', unbound_score_frozen)\n",
    "    print('binding_energy:', peptide, binding_energy)\n",
    "    print('binding_energy_frozen:', peptide, binding_energy_frozen)\n",
    "    print('bound_energies:', peptide,\n",
    "          pose.energies().total_energies().weighted_string_of(\n",
    "              scorefxn.weights()))\n",
    "    print('unbound_energies:', peptide,\n",
    "          unbound_pose.energies().total_energies().weighted_string_of(\n",
    "              scorefxn.weights()))\n",
    "\n",
    "    peptide_scores.append(dict(\n",
    "        peptide=peptide,\n",
    "        template=pmhc_template_pdbfile,\n",
    "        bound_score=bound_score,\n",
    "        unbound_score=unbound_score,\n",
    "        unbound_score_frozen=unbound_score_frozen,\n",
    "        binding_energy=binding_energy,\n",
    "        binding_energy_frozen=binding_energy_frozen,\n",
    "    ))\n",
    "\n",
    "\n",
    "    if outfile_prefix is not None:\n",
    "        # save the final pdb file\n",
    "        peptide_for_filename = peptide.replace('[','_')\\\n",
    "                                      .replace(']','_')\\\n",
    "                                      .replace(':','_')\n",
    "        outfile = f'{outfile_prefix}_{peptide_for_filename}.pdb'\n",
    "        pose.dump_pdb(outfile)\n",
    "\n",
    "        # write out the accumulate scores (we keep over-writing the old scores)\n",
    "        outfile = f'{outfile_prefix}_scores.tsv'\n",
    "        pd.DataFrame(peptide_scores).to_csv(outfile, sep='\\t', index=False)\n",
    "\n",
    "    sys.stdout.flush()\n",
    "\n",
    "\n",
    "\n",
    "print('DONE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "91350e1fe73c1121a5035714d4199d62a57b47d634bafd2c26d1962a1de12d3e"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
