{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2021 [Rosetta PyRosetta4.Release.python38.ubuntu 2021.18+release.54b4909cd528ede1d749ea69e8046c244fc797f2 2021-05-04T21:04:43] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "\u001b[0mcore.init: \u001b[0mChecking for fconfig files in pwd and ./rosetta/flags\n",
      "\u001b[0mcore.init: \u001b[0mRosetta version: PyRosetta4.Release.python38.ubuntu r282 2021.18+release.54b4909cd52 54b4909cd528ede1d749ea69e8046c244fc797f2 http://www.pyrosetta.org 2021-05-04T21:04:43\n",
      "\u001b[0mcore.init: \u001b[0mcommand: PyRosetta -ignore_unrecognized_res 1 -include_current -database /home/jwlee/.local/lib/python3.8/site-packages/pyrosetta/database\n",
      "\u001b[0mbasic.random.init_random_generator: \u001b[0m'RNG device' seed mode, using '/dev/urandom', seed=-2076909332 seed_offset=0 real_seed=-2076909332\n",
      "\u001b[0mbasic.random.init_random_generator: \u001b[0mRandomGenerator:init: Normal mode, seed=-2076909332 RG_type=mt19937\n"
     ]
    }
   ],
   "source": [
    "# Porting command line interface over to Jupyter notebook\n",
    "master_dir = '' #'stuff_for_jihoon_2021-06-04/'\n",
    "output_dir = master_dir + 'run5/'\n",
    "\n",
    "peptides_listfile = output_dir + 'peptides_list_Jihoon-run5-KRAS_muts.txt'\n",
    "pmhc_template_pdbfile = master_dir + 'stuff_for_jihoon_2021-06-04/5yxn.pdb'\n",
    "weights_filetag = 'ref2015_cart.wts' # This is the default\n",
    "\n",
    "outfile_prefix = 'run5'\n",
    "\n",
    "### more imports\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import pyrosetta\n",
    "\n",
    "from pyrosetta.rosetta import core, protocols, numeric, basic, utility\n",
    "from pyrosetta.rosetta.utility import vector1_bool as bools\n",
    "\n",
    "# pyrosetta init\n",
    "init_flags = '-ignore_unrecognized_res 1 -include_current'\n",
    "\n",
    "#if args.ex1:\n",
    "#    init_flags += f' -ex1'\n",
    "#if args.ex2:\n",
    "#    init_flags += f' -ex2'\n",
    "\n",
    "pyrosetta.init(init_flags)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "################################################################################\n",
    "# FUNCTIONS\n",
    "################################################################################\n",
    "\n",
    "def mutate_peptide_sequence( peptide_sequence, pose ):\n",
    "    ''' This function mutates the current peptide sequence in pose\n",
    "    to peptide_sequence\n",
    "    '''\n",
    "    old_sequence = pose.chain_sequence(2)\n",
    "    print(f'mutate_peptide_sequence: from {old_sequence} to',\n",
    "          peptide_sequence)\n",
    "\n",
    "    pep_begin = pose.chain_begin(2)\n",
    "    pep_end = pose.chain_end(2)\n",
    "\n",
    "    amino_acids = 'ACDEFGHIKLMNPQRSTVWY'\n",
    "\n",
    "    if all(x in amino_acids for x in peptide_sequence): # all canonical AAs\n",
    "        assert len(peptide_sequence) == pep_end-pep_begin+1\n",
    "\n",
    "        tf = core.pack.task.TaskFactory()\n",
    "\n",
    "        # freeze non-peptide:\n",
    "        op = core.pack.task.operation.PreventRepacking()\n",
    "        for i in range(1,pose.size()+1):\n",
    "            if pose.chain(i) != 2:\n",
    "                op.include_residue(i)\n",
    "        tf.push_back(op)\n",
    "\n",
    "        # force desired sequence at peptide\n",
    "        for i, aa in zip(range(pose.chain_begin(2), pose.chain_end(2)+1),\n",
    "                         peptide_sequence):\n",
    "            op = core.pack.task.operation.RestrictAbsentCanonicalAAS()\n",
    "            op.include_residue(i)\n",
    "            op.keep_aas(aa)\n",
    "            tf.push_back(op)\n",
    "\n",
    "        packer = protocols.minimization_packing.PackRotamersMover()\n",
    "        packer.task_factory(tf)\n",
    "\n",
    "        packer.apply(pose)\n",
    "    else:\n",
    "        peptide_pose = core.pose.Pose()\n",
    "        core.pose.make_pose_from_sequence(\n",
    "            peptide_pose, peptide_sequence, \"fa_standard\")\n",
    "        for i in range(len(old_sequence)):\n",
    "            old_rsd = pose.residue(pep_begin+i)\n",
    "            new_rsd = peptide_pose.residue(1+i)\n",
    "            if old_rsd.name() != new_rsd.name():\n",
    "                print('residue change:', old_rsd.name(), new_rsd.name())\n",
    "                pose.replace_residue(pep_begin+i, new_rsd, True)\n",
    "\n",
    "\n",
    "def fastrelax_peptide(\n",
    "        scorefxn,\n",
    "        nrepeats,\n",
    "        peptide_positions,\n",
    "        neighbor_positions,\n",
    "        pose\n",
    "):\n",
    "    ''' This function optimizes the rosetta energy of the mhc-peptide complex\n",
    "    It allows the amino acid side chains to change and all atoms in the peptide\n",
    "    and nearby to move a small amount\n",
    "    '''\n",
    "    # movemap:\n",
    "    mm = pyrosetta.rosetta.core.kinematics.MoveMap()\n",
    "    mm.set_bb(False)\n",
    "    mm.set_chi(False)\n",
    "    mm.set_jump(False)\n",
    "\n",
    "    for i in peptide_positions:\n",
    "        mm.set_bb(i, True)\n",
    "        mm.set_chi(i, True)\n",
    "\n",
    "    for i in neighbor_positions:\n",
    "        mm.set_chi(i, True)\n",
    "\n",
    "    print('fastrelax_peptide:: adding chi flex at',\n",
    "          len([x for x in neighbor_positions\n",
    "               if x not in set(peptide_positions)]))\n",
    "\n",
    "    fr = protocols.relax.FastRelax(scorefxn_in=scorefxn,\n",
    "                                   standard_repeats=nrepeats)\n",
    "    fr.cartesian(True)\n",
    "    fr.set_movemap(mm)\n",
    "    fr.set_movemap_disables_packing_of_fixed_chi_positions(True)\n",
    "\n",
    "    # For non-Cartesian scorefunctions, use \"dfpmin_armijo_nonmonotone\"\n",
    "    fr.min_type(\"lbfgs_armijo_nonmonotone\")\n",
    "    fr.apply(pose)\n",
    "\n",
    "\n",
    "def unbind_peptide(pose):\n",
    "    ''' This function slides the peptide away from the MHC so we can\n",
    "    compute an unbound energy\n",
    "    '''\n",
    "    posl = range(1,pose.size()+1)\n",
    "\n",
    "    chains = np.array([pose.chain(x) for x in posl])\n",
    "    calphas = np.array([pose.residue(x).xyz(\"CA\") for x in posl])\n",
    "\n",
    "    mhc_cen = np.mean(calphas[chains==1], axis=0)\n",
    "    pep_cen = np.mean(calphas[chains==2], axis=0)\n",
    "\n",
    "    trans = pep_cen - mhc_cen\n",
    "    trans = numeric.xyzVector_double_t(trans[0], trans[1], trans[2])\n",
    "\n",
    "    j = pose.jump(1)\n",
    "    j.translation_along_axis(pose.conformation().upstream_jump_stub(1), trans,\n",
    "                             25.0)\n",
    "\n",
    "    pose.set_jump(1, j)\n",
    "\n",
    "\n",
    "\n",
    "def find_neighbors(core_positions, pose, heavyatom_distance_threshold = 6.0):\n",
    "    ''' This function finds MHC residues that are nearby the peptide\n",
    "    They will be allowed to move during the energy optimization\n",
    "    '''\n",
    "    # include all the 'core' positions as neighbors (of themselves, e.g.)\n",
    "    nbr_positions = set(core_positions)\n",
    "\n",
    "    posl = range(1, pose.size()+1)\n",
    "\n",
    "    for i in posl:\n",
    "        rsd1 = pose.residue(i)\n",
    "        if rsd1.is_virtual_residue():\n",
    "            continue\n",
    "        for j in core_positions:\n",
    "            rsd2 = pose.residue(j)\n",
    "            if ( rsd1.nbr_atom_xyz().distance_squared( rsd2.nbr_atom_xyz() ) <=\n",
    "                 ( rsd1.nbr_radius() + rsd2.nbr_radius() +\n",
    "                   heavyatom_distance_threshold )**2):\n",
    "                nbr_positions.add(i)\n",
    "                break\n",
    "    return nbr_positions\n",
    "\n",
    "\n",
    "def read_pmhc_pose(filename, special=None):\n",
    "    # 5yxn has the 5th chain as the \n",
    "    ''' This function reads a PDB file and deletes everything except the\n",
    "    MHC peptide binding domain and the peptide\n",
    "\n",
    "    Three possibilities for the input pdb:\n",
    "    1) it has 3 chains: MHC, B2M, peptide\n",
    "    2) it has 5 chains: MHC, B2M, peptide, TCRa, TCRb\n",
    "    3) it has 2 chains: MHC, peptide\n",
    "\n",
    "    '''\n",
    "    pmhc_pose = pyrosetta.pose_from_pdb(filename)\n",
    "\n",
    "    # delete second domain of MHC\n",
    "    if pmhc_pose.chain_end(1) >= 180:\n",
    "        pmhc_pose.delete_residue_range_slow(180, pmhc_pose.chain_end(1))\n",
    "        core.pose.add_variant_type_to_pose_residue(\n",
    "            pmhc_pose, core.chemical.UPPER_TERMINUS_VARIANT,\n",
    "            pmhc_pose.chain_end(1));\n",
    "\n",
    "    if special=='5yxn':\n",
    "        # TCRa, TCRb, MHC, B2M, peptide (doesn't follow the above order)\n",
    "        \n",
    "        # Delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(4), pmhc_pose.chain_end(4))\n",
    "        # Delete TCR\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(1), pmhc_pose.chain_end(2))\n",
    "    \n",
    "    elif pmhc_pose.num_chains()==3:\n",
    "        # MHC, B2M, peptide\n",
    "        # delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(2), pmhc_pose.chain_end(2))\n",
    "\n",
    "    elif pmhc_pose.num_chains() == 5:\n",
    "        # MHC. B2M, peptide, TCRa, TCRb\n",
    "        # delete B2M\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(2), pmhc_pose.chain_end(2))\n",
    "\n",
    "        # delete TCR\n",
    "        pmhc_pose.delete_residue_range_slow(\n",
    "            pmhc_pose.chain_begin(3), pmhc_pose.size())\n",
    "\n",
    "    assert pmhc_pose.num_chains() == 2\n",
    "    return pmhc_pose"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Functions are defined, so now we use them below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.import_pose.import_pose: \u001b[0mFile 'stuff_for_jihoon_2021-06-04/5yxn.pdb' automatically determined to be of type PDB\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue SER:CtermProteinFull 192\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue ASP:CtermProteinFull 433\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 610\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CE  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NZ  on residue LYS 701\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NE  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CZ  on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NH1 on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NH2 on residue ARG 706\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CE  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  NZ  on residue LYS 728\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 753\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 756\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CB  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CG  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  CD  on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE1 on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OE2 on residue GLU 783\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m missing heavyatom:  OXT on residue VAL:CtermProteinFull 818\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 23 92\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 23 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 92 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 23 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 92 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 215 283\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 215 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 283 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 215 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 283 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 734 789\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 734 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 789 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 734 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 789 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 534 597\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 534 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 597 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 534 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 597 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 135 185\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 135 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 185 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 135 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 185 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 334 399\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 334 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 399 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 334 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 399 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 160 360\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 160 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 360 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 160 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 360 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mFound disulfide between residues 636 692\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 636 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 692 CYS\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 636 CYD\n",
      "\u001b[0mcore.conformation.Conformation: \u001b[0mcurrent variant for 692 CYD\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 610 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 701 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 706 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 728 because of missing atom number 6 atom name  CG\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 753 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 756 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.pack_missing_sidechains: \u001b[0mpacking residue number 783 because of missing atom number 5 atom name  CB\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 107 rotamers at 7 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: deleting jump 3\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: renumbering jump 4 to 3\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: deleting jump 1\n",
      "\u001b[0mcore.kinematics.FoldTree: \u001b[0mdelete_jump_seqpos: renumbering jump 3 to 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mutate_peptide_sequence: from KLVALGINAV to KLVVVGAVGV\n",
      "\u001b[0mprotocols.minimization_packing.PackRotamersMover: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m undefined ScoreFunction -- creating a default one\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 13 rotamers at 10 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating PDInteractionGraph\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  234.695  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  234.695  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -216.613  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1819 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -380.415  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -372.952  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -472.805  0.316206  0.316206  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -472.805  0.316206  0.316206  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -347.992  0.316206  0.316206  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1645 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -354.316  0.316206  0.316206  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -346.09  0.316206  0.316206  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -395.006  0.274015  0.274015  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -395.006  0.274015  0.274015  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -308.065  0.274015  0.274015  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1584 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -308.065  0.274015  0.274015  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -301.21  0.274015  0.274015  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -311.102  0.230932  0.230932  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -311.102  0.230932  0.230932  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -201.835  0.230932  0.230932  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1526 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -201.993  0.230932  0.230932  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -213.616  0.185414  0.185414  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -213.616  -213.616  0.185414  0.185414\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -213.616  0.185414  0.185414  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -213.616  0.185414  0.185414  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -125.275  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -125.275  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -340.192  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1789 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -364.938  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -361.003  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -408.245  0.153958  0.153958  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -408.245  0.153958  0.153958  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -309.872  0.153958  0.153958  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1645 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -313.567  0.153958  0.153958  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -307.478  0.153958  0.153958  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -332.932  0.131451  0.131451  0.154\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -332.932  0.131451  0.131451  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -255.141  0.131451  0.131451  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1585 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -255.141  0.131451  0.131451  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -249.007  0.131451  0.131451  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -255.431  0.12341  0.12341  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -255.431  0.12341  0.12341  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -153.421  0.12341  0.12341  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1526 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -153.421  0.12341  0.12341  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -158.4  0.116192  0.116192  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -158.4  -158.4  0.116192  0.116192\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -158.4  0.116192  0.116192  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -158.4  0.116192  0.116192  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:NtermProteinFull]LVVVGAVGV[VAL:CtermProteinFull]\n",
      "bound_score: -213.61647867691948\n",
      "unbound_score: -158.39995051355623\n",
      "unbound_score_frozen: -125.27529469731195\n",
      "binding_energy: KLVVVGAVGV -55.21652816336325\n",
      "binding_energy_frozen: KLVVVGAVGV -88.34118397960754\n",
      "bound_energies: KLVVVGAVGV  fa_atr: -1678.828 fa_rep:   233.765 fa_sol:  1055.710 fa_intra_rep:     3.204 fa_intra_sol_xover4:    60.260 lk_ball_wtd:   -20.100 fa_elec:  -499.350 hbond_sr_bb:   -73.880 hbond_lr_bb:  -100.256 hbond_bb_sc:   -50.922 hbond_sc:   -47.021 dslf_fa13:    -2.369 omega:    56.787 fa_dun:   604.549 p_aa_pp:   -32.770 yhh_planarity:     0.472 ref:    71.610 rama_prepro:    16.914 cart_bonded:   188.610\n",
      "unbound_energies: KLVVVGAVGV  fa_atr: -1606.237 fa_rep:   223.633 fa_sol:  1027.610 fa_intra_rep:     3.143 fa_intra_sol_xover4:    59.900 lk_ball_wtd:   -22.333 fa_elec:  -471.677 hbond_sr_bb:   -73.903 hbond_lr_bb:  -100.256 hbond_bb_sc:   -32.331 hbond_sc:   -51.509 dslf_fa13:    -2.370 omega:    56.042 fa_dun:   598.477 p_aa_pp:   -33.943 yhh_planarity:     0.334 ref:    71.610 rama_prepro:    16.353 cart_bonded:   179.058\n",
      "mutate_peptide_sequence: from KLVALGINAV to KLVVVGAGGV\n",
      "\u001b[0mprotocols.minimization_packing.PackRotamersMover: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m undefined ScoreFunction -- creating a default one\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 13 rotamers at 10 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating PDInteractionGraph\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  239.162  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  239.162  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -210.904  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1816 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -371.917  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -364.94  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -464.865  0.271462  0.271462  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -464.865  0.271462  0.271462  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -352.626  0.271462  0.271462  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1642 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -359.423  0.271462  0.271462  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -352.248  0.271462  0.271462  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -395.715  0.235617  0.235617  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -395.715  0.235617  0.235617  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -310.991  0.235617  0.235617  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1583 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -310.991  0.235617  0.235617  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -304.31  0.235617  0.235617  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -313.948  0.192146  0.192146  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -313.948  0.192146  0.192146  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -206.381  0.192146  0.192146  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1524 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -206.497  0.192146  0.192146  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -213.897  0.169199  0.169199  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -213.897  -213.897  0.169199  0.169199\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -213.897  0.169199  0.169199  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -213.897  0.169199  0.169199  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -127.993  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -127.993  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -342.424  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1788 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -365.628  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -361.842  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -406.489  0.0837672  0.0837672  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -406.489  0.0837672  0.0837672  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -309.382  0.0837672  0.0837672  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1637 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -313.555  0.0837672  0.0837672  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -307.508  0.0837672  0.0837672  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -331.763  0.0784011  0.0784011  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -331.763  0.0784011  0.0784011  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -254.05  0.0784011  0.0784011  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1586 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -254.05  0.0784011  0.0784011  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -247.922  0.0784011  0.0784011  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -254.135  0.0780328  0.0780328  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -254.135  0.0780328  0.0780328  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -151.979  0.0780328  0.0780328  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1527 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -151.979  0.0780328  0.0780328  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -157.226  0.0798683  0.0798683  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -157.226  -157.226  0.0798683  0.0798683\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -157.226  0.0798683  0.0798683  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -157.226  0.0798683  0.0798683  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:NtermProteinFull]LVVVGAGGV[VAL:CtermProteinFull]\n",
      "bound_score: -213.89654756646658\n",
      "unbound_score: -157.22608902804765\n",
      "unbound_score_frozen: -127.99268210279243\n",
      "binding_energy: KLVVVGAGGV -56.67045853841893\n",
      "binding_energy_frozen: KLVVVGAGGV -85.90386546367415\n",
      "bound_energies: KLVVVGAGGV  fa_atr: -1675.168 fa_rep:   232.452 fa_sol:  1059.350 fa_intra_rep:     3.159 fa_intra_sol_xover4:    59.958 lk_ball_wtd:   -21.094 fa_elec:  -497.915 hbond_sr_bb:   -73.792 hbond_lr_bb:  -100.256 hbond_bb_sc:   -50.619 hbond_sc:   -46.037 dslf_fa13:    -2.368 omega:    56.097 fa_dun:   602.706 p_aa_pp:   -32.921 yhh_planarity:     0.510 ref:    69.765 rama_prepro:    17.323 cart_bonded:   184.953\n",
      "unbound_energies: KLVVVGAGGV  fa_atr: -1605.387 fa_rep:   223.812 fa_sol:  1030.141 fa_intra_rep:     3.115 fa_intra_sol_xover4:    59.660 lk_ball_wtd:   -22.494 fa_elec:  -470.480 hbond_sr_bb:   -73.919 hbond_lr_bb:  -100.256 hbond_bb_sc:   -32.338 hbond_sc:   -50.371 dslf_fa13:    -2.368 omega:    55.918 fa_dun:   596.794 p_aa_pp:   -34.715 yhh_planarity:     0.334 ref:    69.765 rama_prepro:    16.908 cart_bonded:   178.655\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mutate_peptide_sequence: from KLVALGINAV to KLVVVGADGV\n",
      "\u001b[0mprotocols.minimization_packing.PackRotamersMover: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m undefined ScoreFunction -- creating a default one\n",
      "\u001b[0mcore.scoring.ScoreFunctionFactory: \u001b[0mSCOREFUNCTION: \u001b[32mref2015\u001b[0m\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 22 rotamers at 10 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating PDInteractionGraph\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mcore.energy_methods.CartesianBondedEnergy: \u001b[0mCreating new peptide-bonded energy container (285)\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  238.699  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  238.699  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -211.596  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1826 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -374.052  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -366.999  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -465.597  0.303768  0.303768  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -465.597  0.303768  0.303768  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -345.342  0.303768  0.303768  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1648 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -353.745  0.303768  0.303768  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -345.962  0.303768  0.303768  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -388.141  0.27095  0.27095  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -388.141  0.27095  0.27095  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -301.785  0.27095  0.27095  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1595 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -301.785  0.27095  0.27095  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -294.975  0.27095  0.27095  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -310.041  0.192688  0.192688  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -310.041  0.192688  0.192688  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -202.223  0.192688  0.192688  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1536 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -202.348  0.192688  0.192688  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -211.776  0.16188  0.16188  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -211.776  -211.776  0.16188  0.16188\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -211.776  0.16188  0.16188  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -211.776  0.16188  0.16188  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "fastrelax_peptide:: adding chi flex at 121\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repeat  -128.075  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -128.075  0  0  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -341.847  0  0  0.022\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1799 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -365.858  0  0  0.022\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -362.09  0  0  0.02805\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -407.513  0.10493  0.10493  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -407.513  0.10493  0.10493  0.02805\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -311.295  0.10493  0.10493  0.14575\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1656 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -314.557  0.10493  0.10493  0.14575\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -308.391  0.10493  0.10493  0.154\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -332.131  0.10066  0.10066  0.154\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -332.131  0.10066  0.10066  0.154\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -253.956  0.10066  0.10066  0.30745\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1596 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -253.956  0.10066  0.10066  0.30745\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -247.792  0.10066  0.10066  0.31955\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -254.233  0.0999944  0.0999944  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: coord_cst_weight  -254.233  0.0999944  0.0999944  0.31955\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: scale:fa_rep  -151.605  0.0999944  0.0999944  0.55\n",
      "\u001b[0mcore.pack.task: \u001b[0mPacker task: initialize from command line()\n",
      "\u001b[0mcore.pack.pack_rotamers: \u001b[0mbuilt 1534 rotamers at 131 positions.\n",
      "\u001b[0mcore.pack.interaction_graph.interaction_graph_factory: \u001b[0mInstantiating DensePDInteractionGraph\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: repack  -151.605  0.0999944  0.0999944  0.55\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 101 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 164 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 203 BRANCH 1\n",
      "\u001b[0mcore.pose.util: \u001b[0m\u001b[1m[ WARNING ]\u001b[0m Unable to find atom_tree atom for this Rosetta branch connection angle: residue 259 BRANCH 1\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: min  -157.012  0.108374  0.108374  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mMRP: 0  -157.012  -157.012  0.108374  0.108374\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: accept_to_best  -157.012  0.108374  0.108374  0.55\n",
      "\u001b[0mprotocols.relax.FastRelax: \u001b[0mCMD: endrepeat  -157.012  0.108374  0.108374  0.55\n",
      "\u001b[0mprotocols::checkpoint: \u001b[0mDeleting checkpoints of FastRelax\n",
      "annotated_sequence: G[GLY:NtermProteinFull]SH[HIS_D]SMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQRMEPRAPWIEQEGPEYWDGETRKVKAHSQTH[HIS_D]RVDLGTLRGYYNQSEAGSH[HIS_D]TVQRMYGC[CYS:disulfide]DVGSDWRFLRGYH[HIS_D]QYAYDGKDYIALKEDLRSWTAADMAAQTTKH[HIS_D]KWEAAHVAEQLRAYLEGTC[CYS:disulfide]VEWLRRYLENGKETLQRTDAPKTHMTHHAVSDHEATLRC[CYS:disulfide]WALSFYPAEITLTWQRDGEDQTQDTELVETRPAGDGTFQKWAAVVVPSGQEQRYTC[CYS:disulfide]HVQHEGLPKPLTLRWE[GLU:CtermProteinFull]K[LYS:NtermProteinFull]LVVVGADGV[VAL:CtermProteinFull]\n",
      "bound_score: -211.77612985703342\n",
      "unbound_score: -157.01196147902067\n",
      "unbound_score_frozen: -128.07491899634806\n",
      "binding_energy: KLVVVGADGV -54.764168378012755\n",
      "binding_energy_frozen: KLVVVGADGV -83.70121086068536\n",
      "bound_energies: KLVVVGADGV  fa_atr: -1675.485 fa_rep:   231.862 fa_sol:  1056.696 fa_intra_rep:     3.147 fa_intra_sol_xover4:    59.972 lk_ball_wtd:   -22.332 fa_elec:  -490.583 hbond_sr_bb:   -73.779 hbond_lr_bb:  -100.256 hbond_bb_sc:   -52.210 hbond_sc:   -45.117 dslf_fa13:    -2.368 omega:    56.486 fa_dun:   603.356 p_aa_pp:   -32.081 yhh_planarity:     0.478 ref:    66.821 rama_prepro:    17.125 cart_bonded:   186.490\n",
      "unbound_energies: KLVVVGADGV  fa_atr: -1606.654 fa_rep:   224.397 fa_sol:  1032.314 fa_intra_rep:     3.192 fa_intra_sol_xover4:    60.060 lk_ball_wtd:   -22.776 fa_elec:  -472.612 hbond_sr_bb:   -73.905 hbond_lr_bb:  -100.256 hbond_bb_sc:   -33.730 hbond_sc:   -50.326 dslf_fa13:    -2.367 omega:    55.974 fa_dun:   599.380 p_aa_pp:   -33.775 yhh_planarity:     0.329 ref:    66.821 rama_prepro:    16.683 cart_bonded:   180.240\n",
      "DONE\n"
     ]
    }
   ],
   "source": [
    "pmhc_pose = read_pmhc_pose(pmhc_template_pdbfile, special='5yxn')\n",
    "\n",
    "# find neighbor positions\n",
    "peptide_positions = list(range(pmhc_pose.chain_begin(2),\n",
    "                               pmhc_pose.chain_end(2)+1))\n",
    "neighbor_positions = find_neighbors(peptide_positions, pmhc_pose)\n",
    "\n",
    "\n",
    "template_peptide = pmhc_pose.chain_sequence(2)\n",
    "\n",
    "with open(peptides_listfile, 'r') as data:\n",
    "    peptides = [x[:-1] for x in data]\n",
    "\n",
    "scorefxn = pyrosetta.create_score_function(weights_filetag)\n",
    "peptide_scores = []\n",
    "\n",
    "for peptide in peptides:\n",
    "    #if len(peptide) != len(template_peptide)\n",
    "    #: This is already checked in the mutate_peptide_sequence function\n",
    "    #    print(f'ERROR: cant model peptide of mismatched length:',\n",
    "    #          f'{peptide} on {template_peptide}')\n",
    "    #    continue\n",
    "\n",
    "    pose = pmhc_pose.clone()\n",
    "\n",
    "    mutate_peptide_sequence(peptide, pose)\n",
    "\n",
    "    fastrelax_peptide(scorefxn, 1, peptide_positions, neighbor_positions, pose)\n",
    "    bound_score = scorefxn(pose)\n",
    "\n",
    "    unbound_pose = pose.clone()\n",
    "\n",
    "    unbind_peptide(unbound_pose)\n",
    "    unbound_score_frozen = scorefxn(unbound_pose)\n",
    "\n",
    "    fastrelax_peptide(scorefxn, 1, peptide_positions, neighbor_positions,\n",
    "                      unbound_pose)\n",
    "    unbound_score = scorefxn(unbound_pose)\n",
    "\n",
    "    binding_energy = bound_score - unbound_score\n",
    "    binding_energy_frozen = bound_score - unbound_score_frozen\n",
    "\n",
    "\n",
    "    print('annotated_sequence:', pose.annotated_sequence())\n",
    "    print('bound_score:', bound_score)\n",
    "    print('unbound_score:', unbound_score)\n",
    "    print('unbound_score_frozen:', unbound_score_frozen)\n",
    "    print('binding_energy:', peptide, binding_energy)\n",
    "    print('binding_energy_frozen:', peptide, binding_energy_frozen)\n",
    "    print('bound_energies:', peptide,\n",
    "          pose.energies().total_energies().weighted_string_of(\n",
    "              scorefxn.weights()))\n",
    "    print('unbound_energies:', peptide,\n",
    "          unbound_pose.energies().total_energies().weighted_string_of(\n",
    "              scorefxn.weights()))\n",
    "\n",
    "    peptide_scores.append(dict(\n",
    "        peptide=peptide,\n",
    "        template=pmhc_template_pdbfile,\n",
    "        bound_score=bound_score,\n",
    "        unbound_score=unbound_score,\n",
    "        unbound_score_frozen=unbound_score_frozen,\n",
    "        binding_energy=binding_energy,\n",
    "        binding_energy_frozen=binding_energy_frozen,\n",
    "    ))\n",
    "\n",
    "\n",
    "    if outfile_prefix is not None:\n",
    "        # save the final pdb file\n",
    "        peptide_for_filename = peptide.replace('[','_')\\\n",
    "                                      .replace(']','_')\\\n",
    "                                      .replace(':','_')\n",
    "        outfile = f'{outfile_prefix}_{peptide_for_filename}.pdb'\n",
    "        pose.dump_pdb(outfile)\n",
    "\n",
    "        # write out the accumulate scores (we keep over-writing the old scores)\n",
    "        outfile = f'{outfile_prefix}_scores.tsv'\n",
    "        pd.DataFrame(peptide_scores).to_csv(outfile, sep='\\t', index=False)\n",
    "\n",
    "    sys.stdout.flush()\n",
    "\n",
    "\n",
    "\n",
    "print('DONE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'peptides_list_Jihoon-run5-KRAS_muts.txt'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides_listfile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptide"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "91350e1fe73c1121a5035714d4199d62a57b47d634bafd2c26d1962a1de12d3e"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
